network/stream/messages.go

```diff
@@ -225,9 +225,14 @@ func (p *Peer) handleOfferedHashesMsg(ctx context.Context, req *OfferedHashesMsg
 	for i := 0; i < lenHashes; i += HashSize {
 		hash := hashes[i : i+HashSize]
 
-		if wait := c.NeedData(ctx, hash); wait != nil {
+		log.Trace("checking offered hash", "ref", fmt.Sprintf("%x", hash))
```

Comment:
potential spam

Reply:
Yes, but also helpful for debugging. `Trace` is not the default log level, so I don't see a problem with this.